openhands/router/rule_based/impl.py

```diff
@@ -0,0 +1,74 @@
+from openhands.core.config import AgentConfig
+from openhands.core.logger import openhands_logger as logger
+from openhands.core.message import Message
+from openhands.events.action import MessageAction
+from openhands.events.event import Event
+from openhands.llm.llm import LLM
+from openhands.llm.llm_registry import LLMRegistry
+from openhands.router.base import ROUTER_REGISTRY, BaseRouter
+
+
+class MultimodalRouter(BaseRouter):
+    SECONDARY_MODEL_CONFIG_NAME = 'secondary_model'
+    ROUTER_NAME = 'multimodal_router'
+
+    def __init__(
+        self,
+        agent_config: AgentConfig,
+        llm_registry: LLMRegistry,
+    ):
+        super().__init__(agent_config, llm_registry)
+
+        self._validate_model_routing_config(self.llms_for_routing)
+
+        # Primary LLM
+        self.llm = llm_registry.get_llm_from_agent_config('agent', agent_config)
+        # Secondary LLM
+        self.secondary_llm = self.llms_for_routing[self.SECONDARY_MODEL_CONFIG_NAME]
+        self.max_token_exceeded = False
+
+    def get_active_llm(self, messages: list[Message], events: list[Event]) -> str:
+        route_to_primary = False
+        # Handle multimodal input
+        for event in events:
+            if (
+                isinstance(event, MessageAction)
+                and event.source == 'user'
+                and event.image_urls
+            ):
+                logger.info('Image content detected. Routing to the primary model.')
```

Comment:
Actually i think we can maybe do this with messages by looking for stuff like `.type != 'text'` in `.content`?